IPython/core/completer.py

```diff
@@ -929,11 +933,13 @@ class IPCompleter(Completer):
     
     @observe('greedy')
     def _greedy_changed(self, change):
-        """update the splitter and readline delims when greedy is changed"""
+        """update the splitters and readline delims when greedy is changed"""
         if change['new']:
-            self.splitter.delims = GREEDY_DELIMS
+            for s in self.splitters:
+                s.delims = GREEDY_DELIMS
         else:
-            self.splitter.delims = DELIMS
+            for s, delims in zip(self.splitters, DELIMS):
+                s.delims = delims
```

Comment:
greedy completer should be deprecated (or maybe it is already) we may be able to just delete this.